### PR TITLE
feat(pipeline): edge side pipeline watch secret event

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
1. edge side pipeline watch secret event
2. set `cluster key` as edge task token

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |  edge side pipeline watch secret event            |
| 🇨🇳 中文    |    边缘pipeline监听cluster key变更          |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: modules/pipeline/providers/edgepipeline_register/interface.go [hunk 1]
```diff
@@ -139,24 +147,26 @@ func (p *provider) GetEdgePipelineEnvs() apistructs.ClusterDialerClientDetail {
...
-		p.Log.Fatalf("failed to make edge connection headers: %v", err)
-	}
+
 	ep, err := p.parseDialerEndpoint()
 	if err != nil {
 		p.Log.Fatalf("failed to parse dialer endpoint: %v", err)
 	}
 	for {
+		if p.getAccessKey() == "" {
+			continue
```
Comment (sfwn) [inline modules/pipeline/providers/edgepipeline_register/interface.go:164]: I think you should sleep here?

File: modules/pipeline/providers/edgepipeline_register/interface.go [hunk 2]
```diff
@@ -233,3 +243,167 @@ func (p *provider) parseDialerEndpoint() (string, error) {
...
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("create retry watcher error: %v", err)
+	}
+
+	return retryWatcher, nil
+}
+
+func (p *provider) getAccessKey() string {
```
Comment (sfwn) [inline modules/pipeline/providers/edgepipeline_register/interface.go:383]: ClusterAccessKey

File: modules/pipeline/providers/edgepipeline_register/interface.go [hunk 3]
```diff
@@ -233,3 +243,167 @@ func (p *provider) parseDialerEndpoint() (string, error) {
...
+	return ac
+}
+
+func (p *provider) setAccessKey(ac string) {
+	p.Lock()
+	defer p.Unlock()
+	p.Cfg.ClusterAccessKey = ac
+}
+
+func (p *provider) getAccessToken() string {
```
Comment (sfwn) [inline modules/pipeline/providers/edgepipeline_register/interface.go:396]: XXXAccessToken

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/pipeline/providers/edgepipeline_register/interface.go (hunk 1 above)
Comment (chengjoey) [reply modules/pipeline/providers/edgepipeline_register/interface.go:165]: adjusted, add sleep

File: modules/pipeline/providers/edgepipeline_register/interface.go (hunk 2 above)
Comment (chengjoey) [reply modules/pipeline/providers/edgepipeline_register/interface.go]: adjusted

File: modules/pipeline/providers/edgepipeline_register/interface.go (hunk 3 above)
Comment (chengjoey) [reply modules/pipeline/providers/edgepipeline_register/interface.go]: adjusted

**Review by sfwn — APPROVED** [review]

**Pull request merged**